### ISSUE TITLE
pic-create & pic-build fail with colon (:) in path

### ISSUE THREAD

**Issue by s9105947** [ComputationalRadiationPhysics/picongpu#3946]
Dear maintainers,

during the automated creation of PIConGPU build directories (using the ISO 8601 datetime format, e.g. `2021-12-16T11:38:57`) I encountered a rather peculiar bug:

## Description
When the path to a PIConGPU setup contains a colon (`:`), both `pic-create` and `pic-build` fail.

## Steps to reproduce
```
# fails:
pic-create LaserWakefield 13:37

# circumvent the failure:
pic-create LaserWakefield working
mv working 19:84
cd '19:84'
# now this fails:
pic-build
```

## Expected Behavior
Both `pic-create` and `pic-build` execute successfully.

## Actual Behavior
Depending on the name of the created directory `pic-create` runs for a while and ultimately prints:

```
$ pic-create LaserWakefield 13:37
Cloning from [...]/picongpu/share/picongpu/examples/LaserWakefield to 13:37
ssh: connect to host 0.0.0.13 port 22: Connection timed out
rsync: connection unexpectedly closed (0 bytes received so far) [sender]
rsync error: unexplained error (code 255) at io.c(235) [sender=3.1.3]
```

`pic-build` fails with:

```
$ pic-build
build directory: .build
cmake command: cmake  -DCMAKE_INSTALL_PREFIX=.. -DPIC_EXTENSION_PATH=..   -DALPAKA_ACC_GPU_CUDA_ENABLE=ON -DALPAKA_ACC_GPU_CUDA_ONLY_MODE=ON -DALPAKA_CUDA_EXPT_EXTENDED_LAMBDA=ON -DCMAKE_CUDA_ARCHITECTURES="70" [...]/picongpu/include/picongpu

[...]

-- Found MPI_C: /opt/spack-modules/linux-ubuntu20.04-skylake_avx512/gcc-10.2.0/openmpi-4.1.1-sxcmu42e6csmwcucp576clilrtvxzo6m/lib/libmpi.so (found version "3.1")
-- Could NOT find MPI_CXX (missing: MPI_CXX_WORKS)
CMake Error at /opt/spack-modules/linux-ubuntu20.04-skylake_avx512/gcc-10.2.0/cmake-3.20.5-xuqouqage2ikjk7bjzxee55nqp27gmj7/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find MPI (missing: MPI_CXX_FOUND) (found version "3.1")
Call Stack (most recent call first):
  /opt/spack-modules/linux-ubuntu20.04-skylake_avx512/gcc-10.2.0/cmake-3.20.5-xuqouqage2ikjk7bjzxee55nqp27gmj7/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /opt/spack-modules/linux-ubuntu20.04-skylake_avx512/gcc-10.2.0/cmake-3.20.5-xuqouqage2ikjk7bjzxee55nqp27gmj7/share/cmake-3.20/Modules/FindMPI.cmake:1742 (find_package_handle_standard_args)
  /home/troepg12/workspace/picongpu/include/pmacc/PMaccConfig.cmake:301 (find_package)
  CMakeLists.txt:150 (find_package)


-- Configuring incomplete, errors occurred!
```

## Notes
There are probably two effects at play here:

For `pic-create` internally `rsync` is issued, where the colon `:` has a special meaning (separating host from path), and hence is not treated as part of the path.

For `pic-build` I supposed (I have not checked thoroughly) the current path is added into some sort of search path.
In this case, the colon `:` has a special meaning too: it is the separator of the paths that are to be searched.

While it would *theoretically* be possible to always escape colons in such paths, it is probably easier to just **forbid paths from containing colons (:)**.
This could be enforced by `pic-create` and `pic-build`, but this problem is probably rare, s.t. the code can be left as-is. (And this issue is documentation of that behavior.)

While *technically* very little restrictions are placed on paths, I did some digging in the (almighty) IEEE Std. 1003.1-2017 (POSIX), and p. 3063 specifies the `pathchk` utility, described as `pathchk - check whether file names are valid or portable`. In normal operation it accepts the name `13:37`, but when enforcing maximum portability with `-p` it complains about the colon. (Well, it also complains about names longer than 14 chars, so there's that.)

```
$ pathchk 13:37
# returns 0 (success)
$ pathchk -p 13:37
pathchk: nonportable character ‘:’ in file name '13:37'
$ pathchk -p abcdefghijklmno
pathchk: limit 14 exceeded by length 15 of file name component ‘abcdefghijklmno’
```

TL;DR: label as wontfix, close, and hope interested users find this issue

**Comment by PrometheusPi** [comment]
Great catch @s9105947 - thanks for opening this issue :+1:

**Comment by BrianMarre** [comment]
@PrometheusPi still needs to be closed and labeled

**Comment by PrometheusPi** [comment]
As requested I will close the issue labeled with `tools` and `wontfix`.
